esadapter/src/main/java/com/ncc/neon/adapters/es/ElasticsearchResultsConverter.java

```diff
@@ -94,7 +100,26 @@ public static TabularQueryResult convertResults(Query query, SearchResponse resp
         return results;
     }
 
-    private static List<Map<String, Object>> extractHitsFromResults(SearchResponse response) {
+    public static List<Map<String, Object>> getScrolledResults(Scroll scroll, SearchResponse response, RestHighLevelClient client) throws IOException {
+        String scrollId = response.getScrollId();
+        List<Map<String, Object>> allHits = new ArrayList<>();
+        allHits.addAll(ElasticsearchResultsConverter.extractHitsFromResults(response));
+
+        while (response.getHits().getHits() != null && response.getHits().getHits().length > 0) {
+            SearchScrollRequest scrollRequest = new SearchScrollRequest(scrollId);
```

Comment:
@jacklechnercaci Do we know how many hits each scroll request returns?  The [documentation](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/6.8/java-rest-high-search-scroll.html#_initialize_the_search_scroll_context) says that the request size is set with the `SearchSourceBuilder`, which we're setting to the query limit in the [ElasticsearchQueryConverter](https://github.com/NextCenturyCorporation/nucleus-data-server/blob/master/esadapter/src/main/java/com/ncc/neon/adapters/es/ElasticsearchQueryConverter.java#L333-L334) (and I'm guessing that ignores the normal 10K maximum?).  So I believe if we query for 15K results, it looks like this just runs a single scroll request for all 15K.  If this is correct, I'd think it may be better for performance (and also less likely to crash) if we run multiple smaller queries (maybe 1K each).  Unless I'm misunderstanding something?

Reply:
There's a config `index.max_result_window` that is defaulted to 10k for performance reasons and it's suggested not to be modified. This means that the max batch size is 10k, but I found `terminateAfter()` in the SearchSourceBuilder that does 10k batches until it reaches the size set there. This should be performant and it does what we wanted.

Reply:
Ah, OK.  So if I want 15K, it will run a request for 10K and then a request for 5K?  That all seems fine.  I'd be interested to know how one 10K request compares to, say, 10 sequential 1K requests, but it would make sense for the 10K to be faster (less overhead).

Reply:
Yep, that's exactly how it works. From what I've seen online, everyone prefers 10k batches.